pages/profile.tsx

```diff
@@ -27,12 +31,37 @@ export default function Profile(props) {
           profileImage: data.users[0].profile_image,
         });
       },
+    });
+
+  const { loading: userGoalsLoading } = useQuery<FetchUserGoalsDataResponse>(
```

Comment:
you should move all the Goals code into a new component called `GoalsSection` move the query into that component as well as the UI for it. 

That way we only query the API when the GoalsSection component is rendered. If you keep the query in the profile page, we always fetch the users goals, even in cases where we want to hide the section by default.

Moving the query logic into a GoalsSection component, allows us to re-use the component in different parts of the student portal as well.